src/openeo_gfmap/manager/job_manager.py

```diff
@@ -203,45 +271,61 @@ def _update_statuses(self, df: pd.DataFrame):
                 job_metadata["status"] == "finished"
             ):
                 _log.info(
-                    f"Job {job.job_id} finished successfully, queueing on_job_done..."
+                    "Job %s finished successfully, queueing on_job_done...", job.job_id
                 )
                 job_status = "postprocessing"
-                future = self._executor.submit(self.on_job_done, job, row)
+                future = self._executor.submit(self.on_job_done, job, row, _stac_lock)
                 # Future will setup the status to finished when the job is done
-                future.add_done_callback(partial(done_callback, df=df, idx=idx))
-                self._futures.append(future)
-                df.loc[idx, "costs"] = job_metadata["costs"]
-                df.loc[idx, "memory"] = (
-                    job_metadata["usage"]
-                    .get("max_executor_memory", {})
-                    .get("value", None)
-                )
-                df.loc[idx, "cpu"] = (
-                    job_metadata["usage"].get("cpu", {}).get("value", None)
-                )
-                df.loc[idx, "duration"] = (
-                    job_metadata["usage"].get("duration", {}).get("value", None)
+                future.add_done_callback(
+                    partial(
+                        done_callback,
+                        df=df,
+                        idx=idx,
+                    )
                 )
+                self._futures.append(future)
+                if "costs" in job_metadata:
+                    df.loc[idx, "costs"] = job_metadata["costs"]
+                    df.loc[idx, "memory"] = (
+                        job_metadata["usage"]
+                        .get("max_executor_memory", {})
+                        .get("value", None)
+                    )
+
+                else:
+                    _log.warning(
+                        "Costs not found in job %s metadata. Costs will be set to 'None'.",
+                        job.job_id,
+                    )
 
             # Case in which it failed
             if (df.loc[idx, "status"] != "error") and (
                 job_metadata["status"] == "error"
             ):
                 _log.info(
-                    f"Job {job.job_id} finished with error, queueing on_job_error..."
+                    "Job %s finished with error, queueing on_job_error...",
+                    job.job_id,
                 )
                 job_status = "postprocessing-error"
```

Comment:
why is the job status in this case set to a post processing error?  This can also occur on a regular error?

Reply:
This is in the case there is a bug in the user-defined post-job action. Setting it to post-processing error would allow the users to stop the manager, fix their code and re-run the post-job actions without needing to re-do the openeo job. You can see the `resume_postproc` parameter in the manager's constructor